### DISCUSSION TITLE
How can I make a script run only after complete synchronization between the server and clients?

### DISCUSSION THREAD

**Discussion by dvai** [TriliumNext/Trilium#4327] [Q&A] [answered]
I have deployed a server and several clients, and I have several scripts that run when the frontend is started (By using the `#run= frontendStartup`). These scripts are responsible for reading and writing note content.

 Sometimes, there is an issue where outdated note data is obtained when there is not complete synchronization, resulting in the content of new notes being overwritten. 

Is there any way for me to ensure that the scripts only run after complete synchronization of the notes?

**Comment by zerebos** [comment]
Have you tried `waitUntilSynced()` see at the bottom of frontend api https://zadam.github.io/trilium/frontend_api/FrontendScriptApi.html

**Reply by dvai** [reply]
Hi, I tried `waitUntilSynced()`, but it seems to only wait for the synchronization between the backend and frontend, not between the server and client. I added `api.waitUntilSynced()` in my script, but the script still executes before the client is fully synced.

**Reply by zadam** [reply]
Yes, this is indeed synchronization between frontend and backend data model, not between two trilium instances.

**Comment by zadam** [comment, marked as answer]
I've added a new backend API call:

```javascript
api.runOutsideOfSync(() => api.log("This will definitely run outside of sync!"));
```

It just exposes a mutex which is used internally for similar needs.

The name sounds silly to me, new naming ideas are welcome!

**Reply by dvai** [reply]
Thanks, this backendapi might be very helpful!
